thefuck/rules/git_stash.py

```diff
@@ -0,0 +1,7 @@
+def match(command, settings):
+    return ('git' in command.script
+            and 'Please commit or stash them.' in command.stderr)
```

Comment:
Could this be tweaked to work for switching branches too?  That error message is `Please, commit your changes or stash them before you can switch branches.` So maybe the `match` could be as simple as `or stash them`.